### DISCUSSION TITLE
Plugin enable/disable/enable detection

### DISCUSSION THREAD

**Discussion by Sachin-M-Desai** [Kong/kong#11048] [Help]
Hello,

We have a requirement to create a table and initialize its fields based on configurations. The table cannot be created at logging opportunity due to performance reasons. Hence we create it only once with a "created" boolean.

However, when a plugin gets disabled and then reenabled with a different configuration, the table needs to change. Is there way for plugin to know that this enable-disable-enable cycle has happened?

I was looking at the created_at field, but that isnt available as part of conf param passed to handlers.

Thank you!

**Comment by shettyh** [comment]
You want to detect plugin config change in the handler ? Then you can refer this 

- https://github.com/Kong/kong/blob/a7bf026a5cf91e376f8cb820e6d8923365f2d581/kong/plugins/acl/handler.lua#L19
- https://github.com/Kong/kong/blob/a7bf026a5cf91e376f8cb820e6d8923365f2d581/kong/plugins/acl/handler.lua#LL49C9-L49C15

**Reply by Sachin-M-Desai** [reply]
Thank you for the quick response. Yes, we would like to perform this in handler.
